query-node/mappings/src/membership.ts

```diff
@@ -83,6 +86,59 @@ async function getOrCreateMembershipSnapshot({ store, event }: EventContext & St
       })
 }
 
+async function saveMembershipExternalResources(
+  store: DatabaseManager,
+  from: MembershipExternalResource[] | undefined,
+  metadata: MembershipMetadata.IExternalResource[] | undefined | null,
+  memberMetadata: MemberMetadata
+): Promise<void> {
+  const externalResource = from ?? metadata?.map(asMembershipExternalResource) ?? []
+  for (const { type, value } of externalResource) {
+    if (type && value) {
+      await store.save<MembershipExternalResource>(new MembershipExternalResource({ type, value, memberMetadata }))
+    }
+  }
+
+  function asMembershipExternalResource(
+    metadata: MembershipMetadata.IExternalResource
+  ): Pick<MembershipExternalResource, 'type' | 'value'> {
+    const type = metadata.type && MembershipMetadata.ExternalResource.ResourceType[metadata.type]
```

Comment:
For failing [test](https://github.com/Joystream/joystream/pull/3862#issuecomment-1158059756), the problem lies here, which also crashed the QN processor with following error

```
ERROR: Stopping the proccessor due to errors: {} name: Error, message: Invalid ResourceType: 0, stack: Error: Invalid ResourceType: 0
```

Here, the code get the corresponding key from `ResourceType` enum after ensuring that `metadata.type` exists, but since type value can also be `0` (corresponds to `EMAIL` option); hence if email option is selected the left operand would be false so the right operand would not be evaluated. 

### Fix
Use builtin utility to check the existence of any metadata property

```suggestion
    const type = isSet(metadata.type) && MembershipMetadata.ExternalResource.ResourceType[metadata.type]
```

Reply:
Thank you so much I was completely stuck here :bow: